tiledb/CMakeLists.txt

```diff
@@ -598,10 +598,10 @@ if (TILEDB_HDFS)
 endif()
 
 # Sanitizer linker flags
-if (SANITIZER)
+if (TILEDB_SANITIZER)
   target_link_libraries(TILEDB_CORE_OBJECTS_ILIB
     INTERFACE
-      -fsanitize=${SANITIZER}
+      -fsanitize=${TILEDB_SANITIZER}
   )
 endif()
```

Comment:
This change only affects some of the sources and targets, not all of them. It only affects the library target, and only the sources in the big source list, which doesn't include anything in the object libraries. At present this is only `baseline`, but that'll change soon.

Reply:
[We actually pass `-fsanitize` globally](https://github.com/TileDB-Inc/TileDB/pull/4515/files#diff-d863ea5136139304facf09da2e8c09a913c3e1d52fd9656c4b70a79248395e80R52) so this block could just be removed?

Reply:
Wait, that was for the compiler, [but we also pass it for the linker](https://github.com/TileDB-Inc/TileDB/pull/4515/files#diff-d863ea5136139304facf09da2e8c09a913c3e1d52fd9656c4b70a79248395e80R69) (MSVC's linker does not have such option). Feel free to remove this block if you see fit. I am on PTO and cannot verify it.

Reply:
I will close the story but added this as a follow up.
https://app.shortcut.com/tiledb-inc/story/38662/follow-up-after-vcpkg-triple-pr